sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/TestCases/FileLines20.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Noncompliant {{This file has 20 lines, which is greater than 10 authorized. Split it into smaller files.}}
```

Comment:
You could amend this test file to make the tests slightly more comprehensive with:
* a block comment / doc comment, and 
* a line that contains both code and a comment

Out of interest, do you know if a completely empty file has a syntax tree?

Reply:
I have added those cases onto the `FileLines9.cs` test case.

Reply:
It's not a big deal, but it isn't possible to tell from the sample code in _FileLines9.cs_ what happens when a line has both code and a comment. The line might or might not be counted, but either way the LOC is still well under the maximum. The only way to check that lines that contain code and a comment are correctly counted would be with a non-compliant example, where the (code + comment) line is included in the LOC reported for the file.